metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/utils/OsCommandHelper.java

```diff
@@ -126,33 +130,38 @@ public static Map<String, File> createOsCommandEmbeddedFiles(
 	public static File createTempFileWithEmbeddedFileContent(
 		final EmbeddedFile embeddedFile,
 		final SudoInformation sudoInformation,
-		Function<String, File> tempFileCreator
+		final BiFunction<String, String, File> tempFileCreator
 	) throws IOException {
-		final String extension = embeddedFile.getType() != null ? "." + embeddedFile.getType() : EMPTY;
+		final String extension = embeddedFile.getFileExtension();
+		final String baseName = embeddedFile.getBaseName();
 
-		final File tempFile = tempFileCreator.apply(extension);
+		final File tempFile = tempFileCreator.apply(baseName, extension);
+		final Path tempFilePath = Paths.get(tempFile.getAbsolutePath());
+		final String contentAsString = embeddedFile.getContentAsString();
 
-		try (
-			BufferedWriter bufferedWriter = Files.newBufferedWriter(
-				Paths.get(tempFile.getAbsolutePath()),
-				StandardCharsets.UTF_8
-			)
-		) {
-			bufferedWriter.write(replaceSudo(embeddedFile.getContent(), sudoInformation));
+		// Should we replace SUDO commands?
+		if (contentAsString.contains("%{SUDO:")) {
+			try (BufferedWriter bufferedWriter = Files.newBufferedWriter(tempFilePath, StandardCharsets.UTF_8)) {
+				bufferedWriter.write(replaceSudo(contentAsString, sudoInformation));
+			}
+		} else {
+			Files.write(tempFilePath, embeddedFile.getContent());
 		}
+
 		return tempFile;
 	}
 
 	/**
 	 * Create a temporary file with the given extension.<br>
 	 * The temporary file name is prefixed with "SEN_Embedded_"
 	 *
+	 * @param baseName  Base name of the file. (File name without extension).
 	 * @param extension File's name suffix (e.g. .bat)
 	 * @return {@link File} instance
 	 */
-	public static File createEmbeddedTempFile(final String extension) {
+	public static File createEmbeddedTempFile(final String baseName, final String extension) {
 		try {
-			return File.createTempFile("SEN_Embedded_", extension);
+			return File.createTempFile("SEN_Embedded_" + baseName, extension);
```

Comment:
Yes, maybe we need `metricshub_` as prefix.
The reason we have this prefix is to easily identify the files in case of an issue. The user or support can then efficiently clean everything in the user's directory, and possibly even on a production machine where files could fill up the filesystem, which could become critical.